### PR TITLE
update: Change Crossplane lab to use DynamoDB instead of RDS

### PR DESCRIPTION
**Pull request by 7navyasa**
#### What this PR does / why we need it:
This PR: 
- Adds functionality to provision a DynamoDB table in AWS using Crossplane instead of an RDS instance, uses the Carts component of the Sample application, updates workshop/lab steps and adds new screenshots.
- Replaces the old Crossplane AWS provider with the Crossplane Upbound AWS provider

#### Which issue(s) this PR fixes:

Fixes # https://github.com/aws-samples/eks-workshop-v2/issues/625
#### Quality checks

- [x] My content adheres to the style guidelines
- [ ] I ran `make test module="<module>"` it was successful (see https://github.com/aws-samples/eks-workshop-v2/blob/main/docs/automated_tests.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by niallthomson** [comment]
@7navyasa this seems to have the same issue as the ACK rework did with a broken link, this was the fix

https://github.com/aws-samples/eks-workshop-v2/pull/712/commits/4710331034bdb65aa94199c36be461e38d36e9b2

File: website/docs/automation/controlplanes/crossplane/resources.md [hunk 1]
```diff
@@ -3,106 +3,127 @@ title: "Managed Resources"
...
-The EC2 security group above allows any traffic from the CIDR range of the VPC used by the EKS cluster. This has been done to keep the example clear and understandable. A more secure approach would be to use [Security Groups for Pods](../../../networking/security-groups-for-pods/index.md) to allow traffic from specific pods.
+Fortunately, we have a handy one-liner to help with this process. Run the below:
 
-:::
+```bash
 
-Next, we want the RDS database to use the private subnets in our VPC. We'll create a `database.aws.crossplane.io.DBSubnetGroup` which selects the appropriate subnet IDs:
+$ eksctl create iamserviceaccount --name carts-crossplane \
+  --namespace carts --cluster $EKS_CLUSTER_NAME \
+  --role-name carts-dynamodb-role \
```
Comment (niallthomson) [inline website/docs/automation/controlplanes/crossplane/resources.md:23]: Please use EKS_CLUSTER_NAME in the role name https://github.com/aws-samples/eks-workshop-v2/blob/61ba415ebfe731ee9cc7d7e662e9904494a5b8c9/website/docs/automation/controlplanes/ack/provision-resources.md?plain=1#L22

File: manifests/modules/automation/controlplanes/crossplane/managed/table.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+apiVersion: dynamodb.aws.upbound.io/v1beta1
+kind: Table
+metadata:
+  name: items
+  labels:
+    testing.upbound.io/example-name: dynamodb
+  annotations: 
+    crossplane.io/external-name: items
```
Comment (niallthomson) [inline manifests/modules/automation/controlplanes/crossplane/managed/table.yaml:8]: This name will have to align with the cluster name, so should be something like `${EKS_CLUSTER_NAME}-carts-crossplane`

File: manifests/modules/automation/controlplanes/crossplane/compositions/claim/claim.yaml [hunk 3]
```diff
@@ -1,11 +1,13 @@
 apiVersion: awsblueprints.io/v1alpha1
-kind: RelationalDatabase
+kind: DynamoDBTable
 metadata:
-  name: catalog-composition
-  namespace: catalog
+  name: items
```
Comment (niallthomson) [inline manifests/modules/automation/controlplanes/crossplane/compositions/claim/claim.yaml:4]: This name will have to align with the cluster name, so should be something like `${EKS_CLUSTER_NAME}-carts-crossplane`

**Review by niallthomson — CHANGES_REQUESTED** [review]

**Comment by niallthomson** [comment]
This commit shows the changes I made to ACK https://github.com/aws-samples/eks-workshop-v2/pull/712/commits/61ba415ebfe731ee9cc7d7e662e9904494a5b8c9

File: website/docs/automation/controlplanes/crossplane/resources.md (hunk 1 above)
Comment (7navyasa) [reply website/docs/automation/controlplanes/crossplane/resources.md]: EKS_CLUSTER_NAME used for the role name parameter

File: manifests/modules/automation/controlplanes/crossplane/managed/table.yaml (hunk 2 above)
Comment (7navyasa) [reply manifests/modules/automation/controlplanes/crossplane/managed/table.yaml]: Updated

File: manifests/modules/automation/controlplanes/crossplane/compositions/claim/claim.yaml (hunk 3 above)
Comment (7navyasa) [reply manifests/modules/automation/controlplanes/crossplane/compositions/claim/claim.yaml]: Updated

**Pull request merged**